ocaml/typing/typecore.ml

```diff
@@ -3662,16 +3691,25 @@ let type_pattern_approx env spat ty_expected =
       end;
   | _ -> ()
 
+and type_pattern_approx_jane_syntax ~env ~ty_expected
+  : Jane_syntax.Pattern.t -> _ = function
+  | Jpat_local (Lpat_local spat) ->
+      type_pattern_approx ~explicitly_local:true env spat ty_expected
+  | Jpat_immutable_array (Iapat_immutable_array _)
+  | Jpat_unboxed_constant (Float _ | Integer _) -> ()
+
+let type_pattern_approx = type_pattern_approx ~explicitly_local:false
```

Comment:
Can we instead make this choice at the callsites of `type_pattern_approx`? That way we don't need this ancillary definition.

Reply:
The idea is that it's _never_ correct to call it with `~explicitly_local:true`, because that can only happen right under a `local_`.  That said, there's also exactly one call site, and it's 31 lines below, so if you feel strongly I can make this change.

Reply:
In this specific case, I think inlining is a clearer choice (but I agree that if this grew many more callsites somehow, we'd want to revisit.)